### PR TITLE
S3 Bam Restoration Feature

### LINKED ISSUE #171: Reinstate BAM restore from archival process
**Context:**

- Also. When applicable, revisit requirement analysis and potentially improve a better file restore procedure / UX flow

### PR DESCRIPTION
**Pull request by williamputraintan**
- Added component to Restore BAM from data action button
- When open archived BAM from `IGV.js` will trigger warning
- Break DataAction big component to a smaller components

Screenshot 2023-02-14 at 12 59 43
Screenshot 2023-02-14 at 13 01 44
Screenshot 2023-02-14 at 13 03 04

Fix #171

### CODE REVIEW

**Comment by victorskl** [comment]
I will review this soon, Will.

**Comment by williamputraintan** [comment]
> I will review this soon, Will.

No rush 😉

**Review by victorskl — APPROVED** [review]
Great effort porting this back and improving it, Will. Thank you!

File: src/api/s3.ts [hunk 1]
```diff
@@ -69,16 +69,16 @@ export async function getS3Status(s3Id: string | number): Promise<S3StatusData>
...
     }
 
-    const isRestoring = restoreStatus.includes('true');
+    const isRestoring = restoreStatus?.includes('true');
     if (isRestoring) {
       return S3StatusData.RESTORING;
     }
 
-    const expiryChunk = restoreStatus.slice(25);
+    const expiryChunk = restoreStatus?.slice(25);
```
[the remark this answers is not in the document]
Comment (williamputraintan) [reply src/api/s3.ts:81]: Hmm I suppose it just to get the expiry chunk out of the string 😅

**Pull request merged**